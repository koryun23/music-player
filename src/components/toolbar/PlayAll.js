import React from "react";
import "../../css/toolbar/Toolbar.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCaretDown, faPlay } from "@fortawesome/free-solid-svg-icons";
import { useState } from "react";


export default function PlayAll(props) {

    const [isDropdownOpen, setIsDropdownOpen] = useState(false);

    const clickDropdown = () => {
        setIsDropdownOpen(!isDropdownOpen);
    }

    return (
        <div className="dropdown dropdown-div play-all-button" onClick={clickDropdown}>
            <button className="button-description btn btn-light dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                <FontAwesomeIcon icon={faPlay} size="lg" className="play-icon"/>
                <b>Play All</b>
            </button>
            <div className={isDropdownOpen ? "dropdown-menu show" : "dropdown-menu"} aria-labelledby="dropdownMenuButton">
                <a href="#" className="dropdown-item">Play Sequentially</a>
                <a href="#" className="dropdown-item">Play Shuffled</a>
            </div>
        </div>
    );
}