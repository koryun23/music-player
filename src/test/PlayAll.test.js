import renderer from "react-test-renderer";
import PlayAll from "../components/toolbar/PlayAll";

it('renders correctly', () => {
    const component = renderer.create(<PlayAll />)
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('play all dropdown works correctly', () => {
    const component = renderer.create(<PlayAll />)
    let tree = component.toJSON();
    console.log(tree.props);
    renderer.act(() => {
        tree.props.onClick();
    })
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    renderer.act(() => {
        tree.props.onClick();
    })
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})