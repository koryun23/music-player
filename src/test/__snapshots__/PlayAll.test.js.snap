// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown works correctly 1`] = `
<div
  className="dropdown dropdown-div play-all-button"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-play fa-lg play-icon"
      data-icon="play"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Play All
    </b>
  </button>
  <div
    aria-labelledby="dropdownMenuButton"
    className="dropdown-menu show"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Repeat After Queue Ends
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Play Queue Only Once
    </a>
  </div>
</div>
`;

exports[`dropdown works correctly 2`] = `
<div
  className="dropdown dropdown-div play-all-button"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-play fa-lg play-icon"
      data-icon="play"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Play All
    </b>
  </button>
  <div
    aria-labelledby="dropdownMenuButton"
    className="dropdown-menu"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Repeat After Queue Ends
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Play Queue Only Once
    </a>
  </div>
</div>
`;

exports[`play all dropdown works correctly 1`] = `
<div
  className="dropdown dropdown-div play-all-button"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-play fa-lg play-icon"
      data-icon="play"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Play All
    </b>
  </button>
  <div
    aria-labelledby="dropdownMenuButton"
    className="dropdown-menu show"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Repeat After Queue Ends
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Play Queue Only Once
    </a>
  </div>
</div>
`;

exports[`play all dropdown works correctly 2`] = `
<div
  className="dropdown dropdown-div play-all-button"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-play fa-lg play-icon"
      data-icon="play"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Play All
    </b>
  </button>
  <div
    aria-labelledby="dropdownMenuButton"
    className="dropdown-menu"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Repeat After Queue Ends
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Play Queue Only Once
    </a>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="dropdown dropdown-div play-all-button"
  onClick={[Function]}
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="button-description btn btn-light dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-play fa-lg play-icon"
      data-icon="play"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M73 39c-14.8-9.1-33.4-9.4-48.5-.9S0 62.6 0 80V432c0 17.4 9.4 33.4 24.5 41.9s33.7 8.1 48.5-.9L361 297c14.3-8.7 23-24.2 23-41s-8.7-32.2-23-41L73 39z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <b>
      Play All
    </b>
  </button>
  <div
    aria-labelledby="dropdownMenuButton"
    className="dropdown-menu"
  >
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Repeat After Queue Ends
    </a>
    <a
      className="dropdown-item"
      href="/"
      onClick={[Function]}
    >
      Play Queue Only Once
    </a>
  </div>
</div>
`;
